drivers/rpi3/gpio/rpi3_gpio.h

```diff
@@ -0,0 +1,22 @@
+#ifndef __RPI3_GPIO_H__
+#define __RPI3_GPIO_H__
+
+#include <gpio.h>
+#include <stdint.h>
+
+typedef struct rpi3_gpio_params {
+	uintptr_t       reg_base;
+} rpi3_gpio_params_t;
+
+void rpi3_gpio_init(rpi3_gpio_params_t *params);
+int rpi3_gpio_get_select(int gpio);
+void rpi3_gpio_set_select(int gpio, int fsel);
+
```

Comment:
You need to add defines here for the GPIO functions:
```
#define RPI3_GPIO_FUNC_INPUT	U(0)
#define RPI3_GPIO_FUNC_OUTPUT	U(1)
#define RPI3_GPIO_FUNC_ALT0	U(4)
#define RPI3_GPIO_FUNC_ALT1	U(5)
#define RPI3_GPIO_FUNC_ALT2	U(6)
#define RPI3_GPIO_FUNC_ALT3	U(7)
#define RPI3_GPIO_FUNC_ALT4	U(3)
#define RPI3_GPIO_FUNC_ALT5	U(2)
```

Then you can change the comments in your functions:
```
 * @param fsel The selection of pinmux. 0: input, 1: output, 2: alt-5,
 *                                      3: alt-4, 4: alt-0,  5: alt-1,
 *                                      6: alt-2, 7: alt-3
```
And you should replace all magic numbers by the defines.